run/idp-sql/pom.xml

```diff
@@ -40,14 +40,14 @@ limitations under the License.
         <!-- Import dependency management from Spring Boot -->
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-dependencies</artifactId>
-        <version>3.1.5</version>
+        <version>2.7.17</version>
         <type>pom</type>
         <scope>import</scope>
       </dependency>
       <dependency>
         <groupId>com.google.cloud</groupId>
         <artifactId>spring-cloud-gcp-dependencies</artifactId>
-        <version>4.7.1</version>
+        <version>3.7.2</version>
```

Comment:
nit: I assume this downgrade is called to support Spring Boot version 2. If not, can you please elaborate the reason for this downgrade?

Reply:
It wouldn't compile in lower versions of java without changing this as well. Full notes in bug